common/membership/grpc_resolver.go

```diff
@@ -27,54 +27,99 @@ package membership
 import (
 	"errors"
 	"fmt"
+	"net/url"
+	"strings"
 	"sync"
-	"sync/atomic"
 
 	"go.uber.org/fx"
 	"google.golang.org/grpc/resolver"
 
 	"go.temporal.io/server/common/primitives"
 )
 
-// GRPCResolver is an empty type used to enforce a dependency using fx so that we're guaranteed to have initialized
-// the global builder before we use it.
-type GRPCResolver struct{}
+const (
+	grpcResolverScheme = "membership"
+
+	// Go's URL parser allows ~ in hostnames
+	delim = "~"
+)
+
+type (
+	// grpcBuilder implements grpc/resolver.Builder. These should be one of these.
```

Comment:
nit: I'm confused by the wording `These should be one of these.`

Reply:
Oh, I just mean it's a singleton and has to be registered with grpc. I can clarify